### DISCUSSION TITLE
[Question] How to get the geojson of all features added by Leaflet-geoman

### DISCUSSION THREAD

**Discussion by AnubhabMondal** [geoman-io/leaflet-geoman#856] [Questions & Help] [answered]
HI @codeofsumit ,
 What code should I implement to get geojson all the features added by geoman ?

**Comment by Falke-Design** [comment, marked as answer]
You can get all layers from the map and check if they have the pm property and not temporary layers.

```
function generateGeoJson(){
	var fg = map.pm.getGeomanDrawLayers(true); // or getGeomanLayers() <-- "true" make that it returns a FeatureGroup instead of an array
	console.log(fg.toGeoJSON());
}
```

Demo: https://jsfiddle.net/falkedesign/054go8j2/

**Comment by AnubhabMondal** [comment]
@Falke-Design Thank You very much. Your replay is valuable and that works for me. Now I can export geojson value.
  Can u plz give me an idea of how to import [Circlel also] those values into my map for next time? If u share something any code structure that would really very helpful for me.
  Thanks.

**Comment by AnubhabMondal** [comment]
Hi,  @Falke-Design Can you plz share something ..?

**Comment by Falke-Design** [comment]
@AnubhabMondal I created a example for you: https://jsfiddle.net/falkedesign/4fycojbu/ 

This has nothing to do with geoman anymore #close

**Comment by AnubhabMondal** [comment]
Thank You so much @Falke-Design for your help.

**Comment by designcomputer** [comment]
Thank you @Falke-Design!

The example code was very helpful. The only problem I had with it was that the Draw Text was not preserved in the GeoJASON data.

**Reply by Falke-Design** [reply]
I understand your problem but it is the same with Circles and CircleMarkers. They are all only a single point and in the GeoJSON spec there is no difference between the Point-Shapes (Marker, Circle, Text).

What you can do is to loop through each layer and if it is a Text layer you can get the GeoJSON of it (`var geo = layer.toGeoJSON()`)  and then add manually properties `geo.properties.text = layer.pm.getText()`

**Reply by designcomputer** [reply]
Thanks again. I'm sure I can get it working using your example.
